src/pybind/mgr/dashboard/frontend/src/app/shared/directives/update-scope.directive.ts

```diff
@@ -0,0 +1,34 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { PermissionScopeDirective } from './permission-scope.directive';
+
+@Directive({
+  selector: 'input:not([cdNoUpdateScope]), select:not([cdNoUpdateScope]), [cdUpdateScope]'
+})
+export class UpdateScopeDirective implements AfterViewInit {
+  private permissions: Permissions;
+  private service_name: keyof Permissions;
+
+  @Input() disabled: boolean;
+
+  constructor(
+    @Optional() private scope: PermissionScopeDirective,
+    private authStorageService: AuthStorageService,
+    private elementRef: ElementRef
+  ) {
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngAfterViewInit() {
+    if (this.scope !== null) {
+      this.service_name = this.scope.cdScope;
+    }
```

Comment:
`assert()`? We don't really want this directive to be used without the Scope one, right?

Reply:
Just a quick doubt. Isnt asserts are usually used for test files? @epuertat

Reply:
Good question. It basically comes down to another question: what's best in case of an unexpected input, silently dealing with any potentially unexpected input or failing quickly and loudly?

With unit tests you already know the answer, and [with production code](https://digitalmars.com/articles/b14.html)? Here you have to perform a couple a checks to deal with potentially missing data (Scope directive, and scope attribute), but that case should never happen and if it happens it would probably be an indication of an slip.

Reply:
I think keeping the scope as Optional() will imply that this directive should not be used without the Scope, correct me if I am wrong. Also here the if condition is checking to see if the scope is not null. So in a way it is the same as assert() right?

Reply:
Ok, now I see it. My bad. We are applying this directive to a generic input selector, so most of the time no ScopeDirective will be defined in a parent element, right? I'm fine with this then!

Reply:
Great. :)